### ISSUE TITLE
Can't get real path

### ISSUE THREAD

**Issue by LiamDawe** [yquake2/yquake2#724]
Hello! First time trying this project out, having an issue getting it going on Linux installed from the Arch AUR git package.

I've moved all the data files over as the documents say to do so.

When I run it, it just says it cannot get the real path:

> [liam@liam-pc yamagi-quake2]$ '/usr/bin/yamagi-quake2' -datadir '/home/liam/.yq2' 
> 
> Yamagi Quake II v8.00
> =====================
> Client build options:
>  + cURL HTTP downloads
>  + OpenAL audio
>  + Systemwide installation
> Platform: Linux
> Architecture: x86_64
> Byte ordering: little endian
> 
> Cmd_RemoveCommand: force_centerview not added
> Cmd_RemoveCommand: +mlook not added
> Cmd_RemoveCommand: -mlook not added
> Shutting down input.
> recursive shutdown
> Error: Couldn't get realpath for /usr/share/games/quake2

**Comment by Yamagi** [comment]
Interesting. First, a stupid question: Does `/usr/share/games/quake2` exist and is it readable?

**Comment by LiamDawe** [comment]
Well, that would have helped if I had looked wouldn't it, woops.

Nope, doesn't exist. Is that supposed to be there from installing Yamagi?

**Comment by Yamagi** [comment]
> Is that supposed to be there from installing Yamagi?

Yes, it is. When the maintainer builds the package as a systemwide install, they must make sure that the data dir exists. Nevertheless, I think I can work around that. Print an error and continue, see if we can get the gamedata otherwise.

I haven't tested it, but a `mkdir /usr/share/games/quake2` may work around the problem. An empty dir may be enough for the game to continue and read the game data from `/home/liam/.yq2`. You don't even need the `/home/liam/.yq2`, because the config dir is always in the data search path.

**Comment by LiamDawe** [comment]
Yep that was enough. I'll let the maintainer of the Arch AUR git package know.

**Comment by Yamagi** [comment]
Very good. At least yo can start playing :)

**Comment by carstene1ns** [comment]
Before 8.00 one could get away with having an optional shared directory (putting the demo there for example).
Most of the time on a single user system the game files will be in `~/.yq2` anyway, so making this error a warning (and not using the directory then) is the best solution.

**Comment by Yamagi** [comment]
I've restored the old behavior. All search path directories are required, only the `SYSTEMDIR` is optional.

**Comment by j4reporting** [comment]
there is a  semicolon missing  in  src/backends/windows/systens.c  at the end of  function Sys_Realpath()

```
src/backends/windows/system.c:612:13: error: expected ';' before '}' token
  612 |  return true
      |             ^
      |             ;
  613 | }
      | ~
make[1]: *** [Makefile:443: build/client/src/backends/windows/system.o] Error 1

```

**Comment by Yamagi** [comment]
This and a braino in LDFLAGS / LDLIBS handling is fixed with 376d97c54fb7ab107ac45bb3231f36d33bde0eb0 and fb080fc32c3bafac3bfb87fcc357f319b07ee7d8. Next time I'll at least compile test under Windows. :/
